planner/cascades/group.go

```diff
@@ -51,18 +54,46 @@ func (g *Group) Insert(e *GroupExpr) bool {
 	if g.Exists(e) {
 		return false
 	}
-	newEquiv := g.equivalents.PushBack(e)
+
+	operand := GetOperand(e.exprNode)
+	var newEquiv *list.Element
+	mark, hasMark := g.firstExpr[operand]
+	if hasMark {
+		newEquiv = g.equivalents.InsertAfter(e, mark)
+	} else {
+		if operand == OperandAny {
```

Comment:
Will we insert any group expr with operand `OperandAny`?

Reply:
No. `OperandAny` should be specially handled in `Group`.